code/src/vs/workbench/api/browser/mainThreadAuthentication.ts

```diff
@@ -152,6 +152,11 @@ export class MainThreadAuthentication extends Disposable implements MainThreadAu
 		}
 
 		// We may need to prompt because we don't have a valid session
+		if (options.silent) {
+			options.silent = false;
+			options.createIfNone = true;
+		}
+
```

Comment:
@vitaliy-guliy 
I tested this solution - it works for me - I'm able to use Github Copilot Chat extension,
thank you!

At the same time it's still unclear how it works on VS Code side without these changes.
Also I detected some side-effects of the current solution:
- let's say I don't have Copilot and Copilot extensions
- It asks me to `sign in using GitHub` directly after starting a workspace 
- without these changes the system asks to `sign in using GitHub` only when I'm using some command that requires it (git clone for example)

Another use case is - `Sign Out`:
- with these changes the system asks to `Sign In` directly after `Sign Out` for all extensions, like: I do `Sign Out` => 3 dialogs is shown to `Sign in` for 3 extensions

Without these changes:
https://github.com/che-incubator/che-code/assets/5676062/87a7f967-6621-4e26-b52d-04fb0e1e4cb3

With these changes:
https://github.com/che-incubator/che-code/assets/5676062/702d9f6b-29f4-4cf4-9c57-e069f5fe3a13

Reply:
Yes, user is asked to grant the access when the extension wants to get a session.
But is seems user is asked only once at the workspace startup.
It should not be very annoying in case you want to use extension like copilot.
From the other side, clicking the button each time when creating a workspace could make a user a bit nervous.

I created another PR to test a bit different behavior https://github.com/che-incubator/che-code/pull/339

Reply:
> But is seems user is asked only once at the workspace startup.

I also described another use case above:
I want to `Sign Out` => 3 dialogs are shown to `Sign in` for 3 extensions

Reply:
I'm not sure users are practicing signing out in a real life.

Reply:
But yes, it's a case